src/sos/workflow_executor.py

```diff
@@ -1635,7 +1635,7 @@ def run_as_master(self, targets=None, mode=None) -> Dict[str, Any]:
         except KeyboardInterrupt as e:
             if exec_error.errors:
                 failed_steps, pending_steps = dag.pending()
-                if pending_steps:
+                if failed_steps:
```

Comment:
This changes the logic of the code. If `failed_steps` is not used, do `pending_steps = dag.pending()[1]`.